From 32cb711402a8539542035213149647cc6537f4ba Mon Sep 17 00:00:00 2001
From: guanzi008 <20619190+guanzi008@users.noreply.github.com>
Date: Tue, 28 Jul 2026 17:02:00 +0800
Subject: [PATCH] FIXED: generate absolute pkg-config paths

The GNUInstallDirs variables used here are relative by default. Prefix them
with the pkg-config prefix variables so consumers do not receive -Llib and
-Iinclude.
---
 source/CMakeLists.txt | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/source/CMakeLists.txt b/source/CMakeLists.txt
index 0dd77b7..9d8a105 100644
--- a/source/CMakeLists.txt
+++ b/source/CMakeLists.txt
@@ -295,8 +295,9 @@ endif()
 
 if(NOT MSVC)
   # Set pkg-config variables
-  set(PKG_CONFIG_LIBDIR ${XLNT_LIB_DEST_DIR})
-  set(PKG_CONFIG_INCLUDEDIR ${XLNT_INC_DEST_DIR})
+  set(PKG_CONFIG_EXEC_PREFIX "\${prefix}")
+  set(PKG_CONFIG_LIBDIR "\${exec_prefix}/${XLNT_LIB_DEST_DIR}")
+  set(PKG_CONFIG_INCLUDEDIR "\${prefix}/${XLNT_INC_DEST_DIR}")
   set(PKG_CONFIG_LIBS "-L\${libdir} -lxlnt")
   set(PKG_CONFIG_CFLAGS "-I\${includedir}")
 
-- 
2.51.0

